src/util/cos.ts

```diff
@@ -0,0 +1,21 @@
+import type { TRadian } from '../typedefs';
+import { halfPI } from '../constants';
+
+/**
+ * Calculate the cos of an angle, avoiding returning floats for known results
+ * This function is here just to avoid getting 0.999999999999999 when dealing
+ * with numbers that are really 1 or 0.
+ * @static
+ * @memberOf fabric.util
+ * @param {TRadian} angle the angle
+ * @return {Number} the cosin value for angle.
+ */
+export const cos = (angle: TRadian): number => {
+  if (angle === 0) { return 1; }
+  var angleSlice = Math.abs(angle) / halfPI;
+  switch (angleSlice) {
+    case 1: case 3: return 0;
+    case 2: return -1;
+  }
+  return Math.cos(angle);
+};
```

Comment:
i like the idea of one function per file and a file of constants.
I m also fine with a file grouping for example cos, sin, degreeToRadians and radiansToDegree, but not much else.

But since rollup has zero footprint on file numbers for me a function per file gives me  easy accessibility to code, i can find the function both in the search and in the file search and i don't have to scroll files of thousands of lines searching the utils.

Reply:
I don't mind if a file for each of grouping.
We should then put cos, sin ,rad/deg utils into a seprate folder IMO called `trig` or something

Reply:
For top leveI functions is just an aestetic preference, i m not sure if there is a meaningful difference in functionality or speed. If there was i think it would be well chatted on the internet.